examples/BCDR/BackupFromStorageQueue.cs

```diff
@@ -0,0 +1,172 @@
+                    log.LogInformation($"Successfully updated key: {setting.Key} label: {setting.Label}");
+                    updatedKeyLabels.Remove(primaryStoreKeyLabel);
+                }
+            }
+
+            // Delete key labels still present in updatedKeyLabels from secondary store
+            foreach (var keyLabel in updatedKeyLabels)
+            {
+                secondaryAppConfigClient.DeleteConfigurationSetting(keyLabel.Key, keyLabel.Label);
+                log.LogInformation($"Successfully deleted key: {keyLabel.Key} label: {keyLabel.Label} from secondary store.");
+            }
+        }
+
+        [FunctionName("BackupFromStorageQueue")]
+        public static void Run([TimerTrigger("0 */10 * * * *")]TimerInfo myTimer, ILogger log)
+        {
+            try
+            {
+                log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+                
+                // Create storage queue client
+                QueueClient queueClient = new QueueClient(Environment.GetEnvironmentVariable("AzureWebJobsStorage"), "eventgridqueue");
+
+                // Peek to see if there are events in the queue
+                while (queueClient.PeekMessages().Value.Length > 0 )
+                {
+                    Response<QueueMessage[]> messages = queueClient.ReceiveMessages(maxMessages: 32);
+
+                    // extract list of key+labels from event data
+                    HashSet<KeyLabel> updatedKeyLabels = ExtractKeyLabelsFromEvents(messages.Value, log);
+
+                    // If there are any valid App Configuration events, update secondary store
+                    if(updatedKeyLabels.Count > 0)
+                    {
+                        BackupKeyValues(updatedKeyLabels, log);
+                    }
+
+                    // Delete this batch of events from storage queue
+                    foreach (QueueMessage message in messages.Value)
+                    {
+                        log.LogInformation($"Deleting message: {message.MessageId}");
+                        queueClient.DeleteMessage(message.MessageId, message.PopReceipt);
+                    }
+                }
+            }
+            catch (RequestFailedException exception)
+            {
+                log.LogError($"Service request failed from the server.\nStatus: {exception.Status}\nError Message: {exception.Message}");
+            }
+            catch (AggregateException exception) when ((exception as AggregateException)?.InnerExceptions?.All(ex => ex is RequestFailedException) ?? false)
+            {
+                log.LogError($"Service request failed from the server with the following exceptions:");
+                foreach (RequestFailedException ex in exception.InnerExceptions)
+                {
+                    log.LogError($"\nStatus: {ex.Status}\nError Message: {ex.Message}");
+                }
+            }
+        }
+    }
+
+    public class KeyLabel
```

Comment:
This should live in its own file and it does not need to be `public`.

Reply:
done